src/systems/user_commands/UserCommands.cc

```diff
@@ -414,9 +414,81 @@ bool CreateCommand::Execute()
     }
     case msgs::EntityFactory::kLight:
     {
-      // TODO(louise) Support light msg
-      ignerr << "light field not yet supported." << std::endl;
-      return false;
+      auto lightMsg = createMsg->light();
+      // Check if there's already a top-level entity with the given name
+      if (kNullEntity != this->iface->ecm->EntityByComponents(
+          components::Name(lightMsg.name()),
+          components::ParentEntity(lightMsg.parent_id())))
+      {
+        ignwarn << "Light entity named [" << lightMsg.name() << "] already"
+                << " exists. Entity not spawned."
+                << std::endl;
+        return false;
+      }
+
+      sdf::Light sdfLight = convert<sdf::Light>(lightMsg);
+
+      std::string lightType = "point";
+      if (lightMsg.type() == msgs::Light::SPOT)
+        lightType = "spot";
+      if (lightMsg.type() == msgs::Light::DIRECTIONAL)
+        lightType = "directional";
+
+      std::string lightStr = std::string("<sdf version='1.7'>") +
+      "<light type='" + lightType + "' name='" + lightMsg.name() + "'> " +
+        "<cast_shadows>" +
+        std::to_string(lightMsg.cast_shadows()) +
+        "</cast_shadows>" +
+        "<pose>" +
+        std::to_string(sdfLight.RawPose().Pos().X()) + " " +
+        std::to_string(sdfLight.RawPose().Pos().Y()) + " " +
+        std::to_string(sdfLight.RawPose().Pos().Z()) + " " +
+        std::to_string(sdfLight.RawPose().Rot().Roll()) + " " +
+        std::to_string(sdfLight.RawPose().Rot().Pitch()) + " " +
+        std::to_string(sdfLight.RawPose().Rot().Yaw()) +
```

Comment:
nothing to change here but just a note that many ign math classes have the `<<` operator so if you use `std::stringstream`, you can just do `ss << sdfLight.RawPose()`